### PR TITLE
docs: add contributing section and devcontainer spec

### PR DESCRIPTION
**Pull request by rokroskar**
Adds a simple devcontainer to make working on the blog easier + contributing section.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
   "devDependencies": {
     "@docusaurus/module-type-aliases": "^3.0.1",
     "@docusaurus/tsconfig": "^3.0.1",
-    "prettier": "^3.0.3",
+    "prettier": "3.2.5",
```
Comment (leafty) [inline package.json:33]: Why is this being updated?

Comment (rokroskar) [reply package.json:33]: I think I installed it separately because for some reason it wasn't there when I ran `npm install`

Comment (leafty) [reply package.json:33]:
```suggestion
    "prettier": "^3.2.5",
```

and re-run `npm i`

**Review by leafty — APPROVED** [review]
LGTM, thanks

**Pull request merged**